src/django/api/views/boundary.py

```diff
@@ -63,3 +68,50 @@ def get(self, request, id, format=None):
         boundary = get_object_or_404(boundary_set, pk=id)
 
         return Response(BoundaryDetailSerializer(boundary).data)
+
+
+class BoundaryShapeView(APIView):
+    permission_classes = [IsAuthenticated]
+
+    def post(self, request, id, format=None):
+        boundary_set = get_boundary_queryset_for_user(request.user)
+        boundary_set = boundary_set.prefetch_related('submissions')
+        boundary = get_object_or_404(boundary_set, pk=id)
+
+        if boundary.status == BOUNDARY_STATUS.DRAFT:
+            if request.user.role not in [Roles.CONTRIBUTOR, Roles.ADMINISTRATOR]:
+                raise ForbiddenException(
+                    'Only contributors and administrators can edit only drafts.'
+                )
+
+            shape = BoundaryShapeView.get_valid_shape(request.data)
+
+            boundary.latest_submission.shape = shape
+            boundary.latest_submission.save()
+
+        elif boundary.status == BOUNDARY_STATUS.IN_REVIEW:
+            if request.user.role not in [Roles.VALIDATOR, Roles.ADMINISTRATOR]:
+                raise ForbiddenException(
+                    'Only validators and administrators can edit boundaries in review.'
```

Comment:
Don't think we should allow Validators changing the shape at all.

Let's allow Contributors and Administrators to change the shape for Drafts, and just that for now. We can add more permissions in the future.

When the shape is changed after the initial submission, it should be logged in the activity log, but we don't have a clear way of logging shape diffs, so I'd rather not do it right now.

Reply:
I was thinking reviews had their own shape, but after I checking the data model I realized they do not. [fea16bc](https://github.com/azavea/iow-boundary-tool/pull/141/commits/fea16bc559b68e9a83eee8a828fb08e61ff07e7c)